### DISCUSSION TITLE
Option store reactivity vs ref

### DISCUSSION THREAD

**Discussion by Lilja** [vuejs/pinia#1680] [Help and Questions]
As far as I can tell, if you define a state through the option store, your properties will end up as vue's `ref` props. But what if I want it to use `reactive`? Do I need to use the composition API style?

Example:

```typescript
defineStore('counter', {
  state() {
    return { selected: {} } // <--- selected will be a `ref`, but I want it to be a `reactive`.
  }
})
  ```

**Comment by baermathias** [comment]
You can do it like this, but `reactive` is not mentioned in the docs as of right now, not even in the composition API style:
```ts
export const useCounterStore = defineStore('counter', () => {
  const count = ref(0)
  const name = ref('Eduardo')
  const doubleCount = computed(() => count.value * 2)
  function increment() {
    count.value++
  }
  return { count, name, doubleCount, increment }
})
```
In Setup Stores:

- `ref()`s become `state` properties
- `computed()`s become `getters`
- `function()`s become `actions`

https://pinia.vuejs.org/core-concepts/#setup-stores

**Reply by evilgeniuscreative** [reply]
Is this supposed to be in the /stores/somefile.js or in a component?? I thought the stores were supposed to be set up in the /src/stores/somestore.js file??
